src/meili/match_route.cc

```diff
@@ -149,7 +149,11 @@ void cut_segments(const std::vector<MatchResult>& match_results,
                                      [&curr_match](const EdgeSegment& segment) {
                                        return (segment.edgeid == curr_match.edgeid);
                                      });
-    assert(last_segment != segments.cend());
+
+    if (last_segment == segments.cend()) {
+      throw std::logic_error("In meili::cutsegments(), unexpectedly unable to locate target edge.");
+      assert(false);
```

Comment:
this seems strange. you want to crash if we are in debug but in release you want to throw. id say just throw in all cases. it results in a failed map match either way. if we find a failed one that we want to investigate we can hop into the debugger and `catch throw` to see the details at the time of the throw

Reply:
actually no. this assert is dead code, as the coverage test is telling us

Reply:
fair enough... not sure why I did that...